flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -129,6 +129,10 @@ class TaskManager(val connectionInfo: InstanceConnectionInfo,
     override def getValue: Double =
       ManagementFactory.getOperatingSystemMXBean().getSystemLoadAverage()
   })
+  metricRegistry.register("cpuLoad", new Gauge[Double] {
+    override def getValue: Double =
+      ManagementFactory.getOperatingSystemMXBean().asInstanceOf[com.sun.management.OperatingSystemMXBean].getProcessCpuLoad()
```

Comment:
This cast may fail on some JVM implementations. See http://www.oracle.com/technetwork/java/faq-sun-packages-142232.html
Can you surround the metric registration by a try catch block and log a warning in case the cast fails?